File: app/src/main/java/vn/loitp/app/activity/customviews/compas/CompasActivity.java
package vn.loitp.app.activity.customviews.compas;

import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.os.Bundle;

import com.core.base.BaseFontActivity;
import com.core.utilities.LLog;
import com.views.compass.LCompass;
import com.views.compass.CompassListener;

import loitp.basemaster.R;

//https://github.com/arbelkilani/Compass-View?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=6973
public class CompasActivity extends BaseFontActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LCompass LCompass = findViewById(R.id.compass_view);
        LCompass.setListener(new CompassListener() {
            @Override
            public void onSensorChanged(SensorEvent event) {
                LLog.INSTANCE.d(getTAG(), "onSensorChanged : " + event);
            }

            @Override
            public void onAccuracyChanged(Sensor sensor, int accuracy) {
                LLog.INSTANCE.d(getTAG(), "onAccuracyChanged : sensor : " + sensor);
                LLog.INSTANCE.d(getTAG(), "onAccuracyChanged : accuracy : " + accuracy);
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_compas;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/compas/CompasActivity.kt
package vn.loitp.app.activity.customviews.compas

import android.hardware.Sensor
import android.hardware.SensorEvent
import android.os.Bundle
import com.core.base.BaseFontActivity
import com.views.compass.CompassListener
import kotlinx.android.synthetic.main.activity_compas.*
import vn.loitp.app.R

//https://github.com/arbelkilani/Compass-View?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=6973
class CompasActivity : BaseFontActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        compassView.setListener(object : CompassListener {
            override fun onSensorChanged(event: SensorEvent) {
                logD("onSensorChanged : $event")
            }

            override fun onAccuracyChanged(sensor: Sensor, accuracy: Int) {
                logD("onAccuracyChanged : sensor : $sensor")
                logD("onAccuracyChanged : accuracy : $accuracy")
            }
        })
    }

    override fun setFullScreen(): Boolean {
        return false
    }

    override fun setTag(): String? {
        return javaClass.simpleName
    }

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_compas
    }
}
